src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1700,28 +1701,30 @@ private void CalculateIORedirection(bool isWindowsApplication, out bool redirect
             "Performance",
             "CA1822:Mark members as static",
             Justification = "Accesses instance members in preprocessor branch.")]
+        [SkipLocalsInit]
         private bool IsExecutable(string path)
         {
 #if UNIX
             return Platform.NonWindowsIsExecutable(this.Path);
 #else
 
             string myExtension = System.IO.Path.GetExtension(path);
+            string pathext = Environment.GetEnvironmentVariable("PATHEXT");
 
-            var pathext = Environment.GetEnvironmentVariable("PATHEXT");
-            string[] extensionList;
-            if (string.IsNullOrEmpty(pathext))
-            {
-                extensionList = new string[] { ".exe", ".com", ".bat", ".cmd" };
-            }
-            else
-            {
-                extensionList = pathext.Split(';');
-            }
+            ReadOnlySpan<char> extensionList = string.IsNullOrEmpty(pathext)
+                ? ".exe;.com;.bat;.cmd"
+                : pathext;
```

Comment:
I guess nobody change default Windows value of the `PATHEXT`. Today it is `.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL`. So we could use statics like:
```c#
const string PATHEXT_DEFAULT = ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL";
private static string[] _pathextDefault = new string[] { ".COM",".EXE",".BAT",".CMD",".VBS",".VBE",".JS",".JSE",".WSF",".WSH",".MSC",".CPL" }
string pathext = Environment.GetEnvironmentVariable("PATHEXT");
if (PATHEXT_DEFAULT  == pathext)
{}
else
{}